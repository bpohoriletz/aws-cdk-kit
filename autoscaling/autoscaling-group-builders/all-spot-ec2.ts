import * as autoscaling from 'aws-cdk-lib/aws-autoscaling';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import AutoScalingGroupBuilderBase from '../autoscaling-group-builder-base';
import { IAutoScalingGroupBullder } from '../products/autoscaling-group';

export class AllSpotEc2AutoscalingGroupBuilder extends AutoScalingGroupBuilderBase implements IAutoScalingGroupBullder {
  setMixedInstancesPolicy(
    launchTemplate?: ec2.ILaunchTemplate | undefined,
    instanceType?: ec2.InstanceType | undefined,
    policy?: autoscaling.MixedInstancesPolicy | undefined
  ): IAutoScalingGroupBullder {
    this.props.mixedInstancesPolicy = policy || {
      instancesDistribution: {
        onDemandBaseCapacity: 0, // All instances are Spot
        onDemandPercentageAboveBaseCapacity: 0, // 100% Spot instances
      },
      launchTemplate: launchTemplate!,
      launchTemplateOverrides: [{ instanceType: instanceType! || new ec2.InstanceType('t2.micro') }],
    };

    return this;
  }
}
