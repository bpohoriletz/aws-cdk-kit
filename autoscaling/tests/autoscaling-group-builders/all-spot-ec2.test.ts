import { AllSpotEc2AutoscalingGroupBuilder } from '../../autoscaling-group-builders/all-spot-ec2';

describe('new AllSpotEc2AutoscalingGroupBuilder()', () => {
  let instance: AllSpotEc2AutoscalingGroupBuilder;

  beforeEach(() => {
    instance = new AllSpotEc2AutoscalingGroupBuilder();
  });

  it('has correct defaults', () => {
    expect(instance).toMatchSnapshot();
  });
});
