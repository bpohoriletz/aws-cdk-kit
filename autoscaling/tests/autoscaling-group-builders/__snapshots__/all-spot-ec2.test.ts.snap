// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new AllSpotEc2AutoscalingGroupBuilder() has correct defaults 1`] = `
AllSpotEc2AutoscalingGroupBuilder {
  "props": AutoscalingGroupProduct {},
}
`;
